sb/robot/arenas/pirate_islands.py

```diff
@@ -0,0 +1,157 @@
+                                           (-point_dist, point_dist)],
+                                          restitution=0.2,
+                                          friction=0.3)
+        super().__init__(arena)
+
+    @property
+    def surface_name(self):
+        return 'sb/pedestal.png'
+
+class PIArena(Arena):
+    start_locations = [(-3.6, -3.6),
+                       (3.6, 3.6)]
+
+    start_headings = [pi / 2,
+                      -pi / 2]
+
+    def __init__(self, objects=None):
+        super().__init__(objects)
+        self._init_pedestals()
+        self._init_tokens()
+
+    @staticmethod
+    def rotate(pos, ang):
+        """
+        Rotate given position around the origin by the given angle
+        :param pos: position to rotate
+        :param ang: angle to rotate
+        :return:
+        """
+        return cos(ang) * pos[0] - sin(ang) * pos[1], sin(ang) * pos[0] + cos(
+            ang) * pos[1]
+
+    @staticmethod
+    def random_pos_spaced(n, min_dist=0.3):
+        """ Creates a n position which are guaranteed to be at least
+        min_dist away from each other
+        """
+        positions = []
+
+        def dist(a, b):
+            return ((a[0] - b[0]) ** 2 + (a[1] - b[1]) ** 2) ** 0.5
+
+        for _ in range(n):
+            while True:
+                pos = (
+                    0.07 + random.random() * 1.7, 0.07 + random.random() * 1.7)
+                if not positions or min(
+                        [dist(pos, other) for other in positions]) > min_dist:
+                    positions.append(pos)
+                    break
+        return positions
+
+    def _init_tokens(self):
+        token_locations = self.random_pos_spaced(4)
+        for zone_id, token_ids in enumerate(
+                [TOKEN_ZONE_0, TOKEN_ZONE_1, TOKEN_ZONE_2, TOKEN_ZONE_3]):
+            for i, (location, token_id) in enumerate(
+                    zip(token_locations, token_ids)):
+                token = Token(self, marker_id=token_id, damping=5)
+                token.location = self.rotate(
+                    (-(location[0] + 0.15), -(location[1] + 0.15)),
+                    zone_id * (math.pi / 2))
+                token.heading = 0
+                self.objects.append(token)
+
+    def _init_pedestals(self):
+        wall_locations = [(-2, 0), (2, 0), (0, -2), (0, 2)]
+        for x, y in wall_locations:
+            wall = Pedestal(self)
+            wall.location = (x, y)
+            self.objects.append(wall)
+
+    def draw_background(self, surface, display):
+        super().draw_background(surface, display)
+
+        def line(start, end, colour=ARENA_MARKINGS_COLOR,
+                 width=ARENA_MARKINGS_WIDTH):
+            pygame.draw.line(surface, colour,
+                             display.to_pixel_coord(
+                                 start), display.to_pixel_coord(end),
+                             width)
+
+        def line_opposite(start, end, **kwargs):
```

Comment:
What exactly do "opposite" and "symmetric" mean in this context? Could we rename the functions or add docstrings to clarify what is happening here?